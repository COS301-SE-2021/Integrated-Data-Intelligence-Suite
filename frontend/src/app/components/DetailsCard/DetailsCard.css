/*The div of the Entire details card
* The Parent div of all the components inside the details card
    including the card 'head' and 'body'
*
*/
#details_card{
    /*border: 3px solid blue;*/
    /*height: 400px;*/
    height: 100%;
    /*width: ;*/
    border: 0;
    box-shadow: none;
}

/*The Label/Title of each description */
th.ant-descriptions-item.map_1 > div.ant-descriptions-item-container > span.ant-descriptions-item-label{
    background-color: #ffffff !important;
    font-weight: 800 !important;
    font-size: 1.4em !important;
    /*border: 3px solid black;*/
    border:0;
    color: black;
}

/*The actual statistic / number value underneath the Label Title*/
td.ant-descriptions-item.map_1 > div.ant-descriptions-item-container > span.ant-descriptions-item-content{
    font-size: 20px;
    /*border: 3px solid orange;*/
    border:0;
    color: #5873f9;
    background: white;
    box-shadow: none;
}

#details_card > .ant-card-head{
    border:0;
}

