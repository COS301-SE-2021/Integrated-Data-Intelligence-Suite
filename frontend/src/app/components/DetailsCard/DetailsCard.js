import React, { Component } from 'react';
import {
    Card,
    Descriptions,
} from 'antd';

const hideDetailsBar = () =>{
    const detailsbar = document.getElementById('map_card_sidebar');
    detailsbar.style.display = 'none';

    // Change Circle color back to normal
    const clicked_circle = document.getElementsByClassName('chosen_circle');
    clicked_circle[0].style.fill = 'red';
    clicked_circle[0].style.fillopacity = 0.2;
    clicked_circle[0].classList.remove('chosen_circle');
};

class DetailsCard extends React.Component {
    // state = {}

    render() {
        return (
            <>
                <Card
                  id="details_card"
                  title=""
                  extra={<p onClick={hideDetailsBar}>close</p>}
                >

                    <Descriptions
                        // className={'descriptions_div'}
                      layout="vertical"
                      column={1}
                      bordered={false}
                      colon={false}
                    >
                        <Descriptions.Item
                          className="map_1"
                          label="Entity Name"
                        >
                            Individual
                        </Descriptions.Item>

                        <Descriptions.Item
                          className="map_1"
                          label="Entity Type"
                        >
                            Strong
                        </Descriptions.Item>

                        <Descriptions.Item
                          className="map_1"
                          label="Average Likes"
                        >
                            46776
                        </Descriptions.Item>
                    </Descriptions>

                </Card>
            </>
        );
    }
}

export default DetailsCard;
