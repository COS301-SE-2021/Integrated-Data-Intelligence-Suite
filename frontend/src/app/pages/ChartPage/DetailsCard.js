import React, {Component} from 'react';
import {Card} from "antd";


class DetailsCard extends React.Component {

    state = {}

    render() {
        return (
            <>
                <Card
                    id={'details_card'}
                    title="Details"
                    extra={<p>Tooltip</p>}
                >


                </Card>
            </>
        );
    }
}

export default DetailsCard;