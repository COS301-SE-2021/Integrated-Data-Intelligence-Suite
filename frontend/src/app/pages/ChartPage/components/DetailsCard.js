import React, {Component} from 'react';
import {
    Card,
    Descriptions
} from "antd";


class DetailsCard extends React.Component {
    state = {}

    render() {
        return (
            <>
                <Card
                    id={'details_card'}
                    title="Details Card Title"
                    extra={<p></p>}
                >

                    <Descriptions
                        // className={'descriptions_div'}
                        layout="vertical"
                        column={1}
                        bordered={false}
                        colon={false}
                    >
                        <Descriptions.Item
                            className={'map_1'}
                            label="Statistic 1"
                        >
                            89%
                        </Descriptions.Item>

                        <Descriptions.Item
                            className={'map_1'}
                            label="Statistic 2"
                        >
                            1810K
                        </Descriptions.Item>

                        <Descriptions.Item
                            className={'map_1'}
                            label="Statistic 3"
                        >
                            Pretoria, Hatfield
                        </Descriptions.Item>
                    </Descriptions>

                </Card>
            </>
        );
    }
}

export default DetailsCard;