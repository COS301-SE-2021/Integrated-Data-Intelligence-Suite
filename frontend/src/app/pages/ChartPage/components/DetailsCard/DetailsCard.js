import React, {Component} from 'react';
import {
    Card,
    Descriptions
} from "antd";

const hideDetailsBar = () =>{
    const detailsbar = document.getElementById("map_card_sidebar")
    detailsbar.style.display="none";
}

class DetailsCard extends React.Component {
    state = {}

    render() {
        return (
            <>
                <Card
                    id={'details_card'}
                    title=""
                    extra={<p onClick={hideDetailsBar}>close</p>}
                >

                    <Descriptions
                        // className={'descriptions_div'}
                        layout="vertical"
                        column={1}
                        bordered={false}
                        colon={false}
                    >
                        <Descriptions.Item
                            className={'map_1'}
                            label="Entity Name"
                        >
                            Individual
                        </Descriptions.Item>

                        <Descriptions.Item
                            className={'map_1'}
                            label="Entity Type"
                        >
                            Strong
                        </Descriptions.Item>

                        <Descriptions.Item
                            className={'map_1'}
                            label="Average Likes"
                        >
                            46776
                        </Descriptions.Item>
                    </Descriptions>

                </Card>
            </>
        );
    }
}

export default DetailsCard;