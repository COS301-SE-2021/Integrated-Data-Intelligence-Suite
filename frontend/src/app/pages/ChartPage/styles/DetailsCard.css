/*The div of the Entire details card
* The Parent div of all the components inside the details card
    including the card 'head' and 'body'
*
*/
#details_card{
    border: 3px solid blue;
    /*height: 400px;*/
    height: 100%;
    /*width: ;*/
}

/*The Label/Title of each description */
th.ant-descriptions-item.map_1 > div.ant-descriptions-item-container > span.ant-descriptions-item-label{
    background-color: #fafafa !important;
    font-weight: 800 !important;
    font-size: 35px !important;
    border: 3px solid black;
    color: black;
}

/*The actual statistic / number value underneath the Label Title*/
td.ant-descriptions-item.map_1 > div.ant-descriptions-item-container > span.ant-descriptions-item-content{
    font-size: 20px;
    border: 3px solid orange;
}

