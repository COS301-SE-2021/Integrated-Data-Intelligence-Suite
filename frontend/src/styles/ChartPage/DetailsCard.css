#details_card{
    border: 3px solid deeppink;
    height: 400px;
    /*width: ;*/
}