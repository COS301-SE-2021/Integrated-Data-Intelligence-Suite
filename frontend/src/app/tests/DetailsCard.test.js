import React from 'react';
import renderer from 'react-test-renderer';
import { unmountComponentAtNode } from 'react-dom';
import DetailsCard from '../components/DetailsCard/DetailsCard';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('DetailsCard renders correctly', () => {
  const tree = renderer
    .create(<DetailsCard />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
