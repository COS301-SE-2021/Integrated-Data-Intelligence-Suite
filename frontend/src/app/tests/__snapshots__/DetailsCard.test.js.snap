// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsCard renders correctly 1`] = `
<div
  className="ant-card ant-card-bordered"
  id="details_card"
>
  <div
    className="ant-card-head"
    style={Object {}}
  >
    <div
      className="ant-card-head-wrapper"
    >
      
      <div
        className="ant-card-extra"
      >
        <p
          onClick={[Function]}
        >
          close
        </p>
      </div>
    </div>
  </div>
  <div
    className="ant-card-body"
    style={Object {}}
  >
    <div
      className="ant-descriptions"
    >
      <div
        className="ant-descriptions-view"
      >
        <table>
          <tbody>
            <tr
              className="ant-descriptions-row"
            >
              <th
                className="ant-descriptions-item map_1"
                colSpan={1}
              >
                <div
                  className="ant-descriptions-item-container"
                >
                  <span
                    className="ant-descriptions-item-label ant-descriptions-item-no-colon"
                    style={Object {}}
                  >
                    Entity Name
                  </span>
                </div>
              </th>
            </tr>
            <tr
              className="ant-descriptions-row"
            >
              <td
                className="ant-descriptions-item map_1"
                colSpan={1}
              >
                <div
                  className="ant-descriptions-item-container"
                >
                  <span
                    className="ant-descriptions-item-content"
                    style={Object {}}
                  >
                    Individual
                  </span>
                </div>
              </td>
            </tr>
            <tr
              className="ant-descriptions-row"
            >
              <th
                className="ant-descriptions-item map_1"
                colSpan={1}
              >
                <div
                  className="ant-descriptions-item-container"
                >
                  <span
                    className="ant-descriptions-item-label ant-descriptions-item-no-colon"
                    style={Object {}}
                  >
                    Entity Type
                  </span>
                </div>
              </th>
            </tr>
            <tr
              className="ant-descriptions-row"
            >
              <td
                className="ant-descriptions-item map_1"
                colSpan={1}
              >
                <div
                  className="ant-descriptions-item-container"
                >
                  <span
                    className="ant-descriptions-item-content"
                    style={Object {}}
                  >
                    Strong
                  </span>
                </div>
              </td>
            </tr>
            <tr
              className="ant-descriptions-row"
            >
              <th
                className="ant-descriptions-item map_1"
                colSpan={1}
              >
                <div
                  className="ant-descriptions-item-container"
                >
                  <span
                    className="ant-descriptions-item-label ant-descriptions-item-no-colon"
                    style={Object {}}
                  >
                    Average Likes
                  </span>
                </div>
              </th>
            </tr>
            <tr
              className="ant-descriptions-row"
            >
              <td
                className="ant-descriptions-item map_1"
                colSpan={1}
              >
                <div
                  className="ant-descriptions-item-container"
                >
                  <span
                    className="ant-descriptions-item-content"
                    style={Object {}}
                  >
                    46776
                  </span>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
